File: K3.jl
using CSV, DataFrames, PrettyTables, Clustering, Plots, StatsPlots, Random

df = CSV.read("dye.csv", DataFrame)
df_group = groupby(df, :class)

df_centers = DataFrame()
rng = MersenneTwister(405)

for class in df_group
    shuffled_class = shuffle(rng, class)
    first_50_rows = shuffled_class[1:50, :]
    
    global X = select(first_50_rows, Not("class"))
    global features = Matrix(X)'
    global result = kmeans(features, 2; maxiter=20, rng)
    df_what_ever = DataFrame(d1 = result.centers'[:, 1],
                             c1 = result.centers'[:, 2],
                             d2 = result.centers'[:, 3],
                             c2 = result.centers'[:, 4],
                             d3 = result.centers'[:, 5],
                             c3 = result.centers'[:, 6])
    append!(df_centers, df_what_ever)
end

pretty_table(Matrix(df_centers))

File: K3.py
import pandas as pd
from sklearn.cluster import KMeans
import numpy as np

df = pd.read_csv("dye.csv")
df_group = df.groupby('class')

df_centers = pd.DataFrame()

for name, group in df_group:
    shuffled_group = group.sample(frac=1, random_state=405)
    first_50_rows = shuffled_group.head(50)
    
    X = first_50_rows.drop(columns=['class'])

    kmeans = KMeans(n_clusters=2, max_iter=20, random_state=405)
    result = kmeans.fit(X)
    
    df_temp = pd.DataFrame({
        'd1': result.cluster_centers_[:, 0],
        'c1': result.cluster_centers_[:, 1],
        'd2': result.cluster_centers_[:, 2],
        'c2': result.cluster_centers_[:, 3],
        'd3': result.cluster_centers_[:, 4],
        'c3': result.cluster_centers_[:, 5]
    })
    df_centers = pd.concat([df_centers, df_temp], ignore_index=True)

print(df_centers)